### DISCUSSION TITLE
Implementation of marp for slide shows

### DISCUSSION THREAD

**Discussion by harrelfe** [quarto-dev/quarto-cli#7377] [Feature Requests]
### Description

I just discovered [`marp`](https://marp.app) for producing beautiful slide shows from simple `markdown` specifications similar to the markdown used in slide presentations with `Quarto`.  `marp` produces self-contained `html` files and also `pptx`, `pdf`, and others.  The reason for using `marp` is that the `html` files it produces are 50 times smaller than those produced by `reveal.js` and 8 times smaller than those produced by `ioslides`, in some simple tests I ran.  The `marp` command line program is a standalone binary that’s not too big.  What about adding it to `Quarto`’s repertoire?

**Comment by harrelfe** [comment]
You can get part of the way there using format: commonmark in the `.qmd` file but you have to do a lot of editing of the resulting `.md` file, such as changing all -------------------... to ---; change - bullet symbols to *; get rid of newlines inserted into bullet text, remove an initial blank line inserted as a result of `include-in-header: marpheader.md` where `marpheader.md` is something like
```
---
theme: gaia
_class: lead
paginate: true
backgroundColor: #fff
header: '![w:150](logo.png)'
_paginate: false     # no page number on title slide
marp: true              # for VS Code editor
---
```

**Reply by mcanouil** [reply]
If the editing obeys some rules, a post-render can be written.
Then you can even create an extension for that.

**Reply by harrelfe** [reply]
Thanks.  The hardest part is that `quarto` rendering into `commonmark` breaks lines within bullet points, making `marp` output not flow correctly.  Easier to fix is a strange problem where bullet lines starting with `*` are changed by `quarto` to `-` which I thought `commonmark` preferred, but this makes `Marp` not treat the bullet point as progressively revealed.

I added a `Marp` template/example [here](https://hbiostat.org/r/marp-template.md) with output [here](https://hbiostat.org/r/marp-template.html).  This did not use `quarto`.

**Reply by mcanouil** [reply]
Note that the "writers" are the ones from Pandoc.

**Comment by mcanouil** [comment]
FYI

- https://github.com/jgm/pandoc/issues/10483
